server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -564,6 +567,8 @@ private static void parseObject(final ParseContext context, ObjectMapper mapper,
     private static void parseArray(ParseContext context, ObjectMapper parentMapper, String lastFieldName, String[] paths)
         throws IOException {
         String arrayFieldName = lastFieldName;
+        context.incrementFieldArrayDepth();
```

Comment:
I think splitting the counter in two is not going to work as expected, it seems to me it is feasible to craft the test case like: `object -> array -> object -> array ....` and basically double the effective depth limit  (depth of objects + depth of arrays).

Reply:
Yes, i agree to that. But the [issue](https://github.com/opensearch-project/OpenSearch/issues/5195) this PR is raised to resolve is to handle unbounded growth of stack usage (and potentially causing stack to overflow) due to recursion while parsing an ill-formed request body. In worst case, the recursion depth can go upto (depth of objects + depth of arrays) before throwing an exception. But it won't cause deep recursion beyond depth=40(default). I kept both the counters separate to keep them independent. Please let me know if you see any issue here.

Reply:
Ok, could you please add test case for combination of `object -> array -> object -> array ....`  so at least we capture this edge case, thanks.

Reply:
added test case for combination of object -> array -> object -> array ....

Reply:
Hi @reta , is there any other comment or feedback on the PR from your side?

Reply:
My apologies, I missed this message, LGTM!